snuba/api.py

```diff
@@ -372,24 +375,24 @@ def parse_and_run_query(dataset, body, timer):
         ], key=lambda priority_and_col: priority_and_col[0])
         if prewhere_candidates:
             prewhere_conditions = [cond for _, cond in prewhere_candidates][:settings.MAX_PREWHERE_CONDITIONS]
-            body['conditions'] = list(filter(lambda cond: cond not in prewhere_conditions, body['conditions']))
+            request.query['conditions'] = list(filter(lambda cond: cond not in prewhere_conditions, request.query['conditions']))
 
     table = dataset.get_schema().get_table_name()
 
-    sql = format_query(dataset, body, table, prewhere_conditions, final)
+    sql = format_query(dataset, request, table, prewhere_conditions, final)
 
     timer.mark('prepare_query')
 
     stats = {
         'clickhouse_table': table,
         'final': final,
-        'referrer': request.referrer,
+        'referrer': http_request.referrer,
```

Comment:
Didn't realize we were getting this directly from the header. Seems like abuse of HTTP.